plugins/modules/mso_schema_site_external_epg_selector.py

```diff
@@ -0,0 +1,251 @@
+  register: query_result
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec, mso_contractref_spec, issubset, mso_expression_spec_ext_epg
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type='str', required=True),
+        template=dict(type='str', required=True),
+        site=dict(type='str', required=True),
+        external_epg=dict(type='str', required=True),
+        selector=dict(type='str'),
+        ip_address=dict(type='str'),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    schema = module.params.get('schema')
+    template = module.params.get('template')
+    site = module.params.get('site')
+    external_epg = module.params.get('external_epg')
+    selector = module.params.get('selector')
+    ip_address = module.params.get('ip_address')
+    state = module.params.get('state')
+
+    mso = MSOModule(module)
+
+    # Get schema_id
+    schema_obj = mso.get_obj('schemas', displayName=schema)
+    if not schema_obj:
+        mso.fail_json(msg="Provided schema '{0}' does not exist".format(schema))
+
+    schema_path = 'schemas/{id}'.format(**schema_obj)
+    schema_id = schema_obj.get('id')
+
+    # Get template
+    templates = [t.get('name') for t in schema_obj.get('templates')]
+    if template not in templates:
+        mso.fail_json(msg="Provided template '{template}' does not exist. Existing templates: {templates}".format(template=template,
+                                                                                                                  templates=', '.join(templates)))
+
+    # Get site
+    site_id = mso.lookup_site(site)
+
+    # Get site_idx
+    sites = [(s.get('siteId'), s.get('templateName')) for s in schema_obj.get('sites')]
+    # Schema-access uses indexes
+    site_idx = sites.index((site_id, template))
+    # Path-based access uses site_id-template
+    site_template = '{0}-{1}'.format(site_id, template)
+
+    payload = dict()
+    op_path = ''
+
+    # Get External EPG
+    ext_epg_ref = mso.ext_epg_ref(schema_id=schema_id, template=template, external_epg=external_epg)
+    external_epgs = [e.get('externalEpgRef') for e in schema_obj.get('sites')[site_idx]['externalEpgs']]
+
+    if ext_epg_ref not in external_epgs:
+        op_path = '/sites/{0}/externalEpgs/-'.format(site_template)
+        payload.update(
+            externalEpgRef=dict(
+                schemaId=schema_id,
+                templateName=template,
+                externalEpgName=external_epg,
+            )
+        )
+        payload.update(l3outDn='')
+    else:
+        external_epg_idx = external_epgs.index(ext_epg_ref)
+
+    # Get Selector
+    if not payload:
```

Comment:
I don't think this condition is necessary because if you remove this line you are still in the previous else statement which is equivalent to "if not payload"